### PR TITLE
Conformance tests for portforward_test

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
This PR aims to identify and label the tests in the portforward_test file.

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
/approve

File: tests/replicaset_test.go [hunk 1]
```diff
@@ -66,19 +63,20 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 	})
 
 	doScale := func(name string, scale int32) {
-
+		pathBytes, err := patch.New(patch.WithReplace("/spec/replicas", scale)).GeneratePayload()
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply tests/replicaset_test.go]: Yep, thanks! Fixed

File: tests/replicaset_test.go [hunk 2]
```diff
@@ -322,30 +321,35 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 		Expect(err).ToNot(HaveOccurred())
 	})
 
-	It("[test_id:1417]should not scale when paused and scale when resume", func() {
+	It("[test_id:1417]should not scale when paused and scale when resume", decorators.Conformance, func() {
 		rs := newReplicaSet()
 		// pause controller
 		By("Pausing the replicaset")
-		_, err := virtClient.ReplicaSet(rs.Namespace).Patch(context.Background(), rs.Name, types.JSONPatchType, []byte("[{ \"op\": \"add\", \"path\": \"/spec/paused\", \"value\": true }]"), v12.PatchOptions{})
+		pathBytes, err := patch.New(patch.WithAdd("/spec/paused", true)).GeneratePayload()
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply tests/replicaset_test.go]: same

File: tests/replicaset_test.go [hunk 3]
```diff
@@ -145,7 +145,7 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 		Entry("[test_id:1406]to five, to six and then to zero replicas", 5, 6),
 	)
 
-	DescribeTable("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:component]should scale with scale subresource", func(startScale int, stopScale int) {
+	DescribeTable("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:component]should scale with scale subresource", decorators.Conformance, func(startScale int, stopScale int) {
```
Comment (xpivarc) [inline tests/replicaset_test.go:148]: There were proposal to deprecate the ReplicaSet in favor of new Pool API, let's defer the Conformance for this.

Comment (fossedihelm) [reply tests/replicaset_test.go:148]: I will drop both replicaset commits.

**Comment by fossedihelm** [comment]
@xpivarc PTAL

File: tests/portforward_test.go [hunk 4]
```diff
@@ -49,7 +49,7 @@ var _ = Describe("[sig-compute]PortForward", decorators.SigCompute, func() {
 		virtClient = kubevirt.Client()
 	})
 
-	It("should successfully open connection to guest", func() {
+	It("should successfully open connection to guest", decorators.Conformance, func() {
```
Comment (xpivarc) [inline tests/portforward_test.go:52]: ` 			libvmi.WithNamespace(testsuite.NamespaceTestDefault),` is smell

Comment (xpivarc) [inline tests/portforward_test.go:52]: The `err    error` should be in `Eventually`

**Review by xpivarc — COMMENTED** [review]
Some notes below

**Comment by fossedihelm** [comment]
@xpivarc Comments addressed, PTAL

**Comment by xpivarc** [comment]
I would approve if `pull-kubevirt-check-tests-for-flakes` would not detect flake :)  Can you investigate why this happens?

File: tests/portforward_test.go [hunk 5]
```diff
@@ -49,19 +46,16 @@ var _ = Describe("[sig-compute]PortForward", decorators.SigCompute, func() {
...
 		vmi := libvmifact.NewFedora(
 			libnet.WithMasqueradeNetworking(),
-			libvmi.WithNamespace(testsuite.NamespaceTestDefault),
 		)
-		libvmops.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 180)
+		vmi = libvmops.RunVMIAndExpectLaunchIgnoreWarnings(vmi, 180)
```
Comment (fossedihelm) [inline tests/portforward_test.go:53]: @xpivarc Previously the vmi was created with `libvmi.WithNamespace(testsuite.NamespaceTestDefault)`, and when we later perform the portforward in the `vmi.Namespace` ns it was working.
Having dropped the  `libvmi.WithNamespace(testsuite.NamespaceTestDefault)`, we need to update our local vmi object with the one retrieved during launch.
@0xFelix FYI

**Comment by fossedihelm** [comment]
@xpivarc It should be solved now :)

File: tests/portforward_test.go (hunk 5 above)
Comment (xpivarc) [reply tests/portforward_test.go:53]: Nice :)

**Review by xpivarc — COMMENTED** [review]
/approve

**Comment by fossedihelm** [comment]
@xpivarc @0xFelix 
This PR received 2 approve, can it get an lgtm? :)

**Comment by xpivarc** [comment]
> @xpivarc @0xFelix This PR received 2 approve, can it get an lgtm? :)

I was just waiting for tests  :)

/lgtm

**Pull request merged**